File: RedSheet.DbEntities/Models/vUser.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vUsers", Schema = "dbo")]
	[Serializable]
	public partial class vUser  
	{
 
	[Required] 
	[MaxLength(100)]
	public string Address { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string City { get; set; } 

	public Nullable<Guid> CompanyId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string CompanyName { get; set; } 
 
	[Required] 
	[MaxLength(150)]
	public string Email { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string FirstName { get; set; } 
 
	[Required] 
	[MaxLength(6)]
	public string Initial { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string LastName { get; set; } 
 
	[Required] 
	[MaxLength(20)]
	public string Mobile { get; set; } 
 
	[Required] 
	[MaxLength(20)]
	public string Office { get; set; } 

	public Nullable<Guid> RequestorId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string State { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string Status { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string Title { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int UserId { get; set; } 
 
	[Required] 
	[MaxLength(10)]
	public string ZipCode { get; set; } 

	public vUser()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-user.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vUser {
    constructor(vUser?: vUser )  {
        let properties = [ "address", "city", "companyId", "companyName", "email", "firstName", "initial", "lastName", "mobile", "office", "requestorId", "state", "status", "title", "userId", "zipCode",];
        for (let property of properties)
            if (vUser && vUser[property])
                this[property] = vUser[property];
    }
 
    @required()
    @maxLength(100)
	address : string =   undefined;
 
    @required()
    @maxLength(50)
	city : string =   undefined;
 
	companyId : string =   undefined;
 
    @required()
    @maxLength(100)
	companyName : string =   undefined;
 
    @required()
    @maxLength(150)
	email : string =   undefined;
 
    @required()
    @maxLength(100)
	firstName : string =   undefined;
 
    @required()
    @maxLength(6)
	initial : string =   undefined;
 
    @required()
    @maxLength(100)
	lastName : string =   undefined;
 
    @required()
    @maxLength(20)
	mobile : string =   undefined;
 
    @required()
    @maxLength(20)
	office : string =   undefined;
 
	requestorId : string =   undefined;
 
    @required()
    @maxLength(100)
	state : string =   undefined;
 
    @required()
    @maxLength(100)
	status : string =   undefined;
 
    @required()
    @maxLength(50)
	title : string =   undefined;
 
	userId : number =   0 ;
 
    @required()
    @maxLength(10)
	zipCode : string =   undefined;


}
